bundler/lib/bundler/plugin.rb

```diff
@@ -342,7 +342,26 @@ def load_plugin(name)
       # done to avoid conflicts
       path = index.plugin_path(name)
 
-      Gem.add_to_load_path(*index.load_paths(name))
+      paths = index.load_paths(name)
+      paths.each do |p|
+        next if File.directory?(p)
+
+        message = <<~MESSAGE
+          Plugin path #{p} does not exist.
+
+          This can happen if the plugin was installed with a different version of Ruby that has since
+          been uninstalled. To fix this, run:
+
+          bundler plugin uninstall #{name}
+
+          If you would like to reinstall the plugin, run:
+
+          bundler plugin install #{name}
+        MESSAGE
+        raise PluginError.new(message)
+      end
+
+      Gem.add_to_load_path(paths)
```

Comment:
@sarahsehr this seems to fails the test (and breaks some test gem installation at the same time). `add_to_load_path` is defined at https://github.com/rubygems/rubygems/blob/c1933bfd4fc7579f679e80599b9ed28791619e24/lib/rubygems.rb#L576-L584 and it expects multiple arguments. To achieve this by passing `paths` array, you need to add splat (*) operator. Changing this to `Gem.add_to_load_path(*paths)` should fix this. :pray: